packages/wonder-blocks-dropdown/src/hooks/use-select-validation.tsx

```diff
@@ -0,0 +1,145 @@
+            (validate && hasValue(value) && !disabled && validate(value)) ||
+            null,
+    );
+
+    const handleValidation = React.useCallback(
+        (newValue?: T) => {
+            // Should not handle validation if it is disabled
+            if (disabled) {
+                return;
+            }
+            if (validate) {
+                const error =
+                    (newValue !== undefined && validate(newValue)) || null;
+                setErrorMessage(error);
+                if (onValidate) {
+                    onValidate(error);
+                }
+                if (error) {
+                    // If there is an error, do not continue with validation related to `required` prop
+                    return;
+                }
+            }
+            if (required) {
+                const requiredString =
+                    typeof required === "string"
+                        ? required
+                        : defaultErrorMessage;
+                const error = hasValue(newValue) ? null : requiredString;
+                setErrorMessage(error);
+                if (onValidate) {
+                    onValidate(error);
+                }
+            }
+        },
+        [disabled, validate, setErrorMessage, onValidate, required],
+    );
+
+    useOnMountEffect(() => {
+        // Only validate on mount if the value is not empty. This is so that
+        // fields don't render an error when they are initially empty
+        if (hasValue(value)) {
+            handleValidation(value);
+        }
+    });
+
+    function onOpenerBlurValidation() {
+        if (!open && required && !hasValue(value)) {
+            // Only validate on opener blur if the dropdown is closed, the field
+            // is required, and no value is selected. This prevents an error state
+            // when the dropdown is opened without a value yet since opening
+            // the dropdown will also trigger the blur event since focus is moved
+            // to the dropdown
+            handleValidation(value);
+        }
+    }
+
+    const onDropdownClosedValidation = () => {
+        if (required && !hasValue(value)) {
+            // If field is required, and no value is selected, validate
+            // We don't check for `!open` because `open` is not updated yet
+            // when this is called
+            handleValidation(value);
+        }
+    };
+
+    const onSelectionValidation = (newValue: T) => {
+        handleValidation(newValue);
+    };
+
+    const onSelectedValuesChangeValidation = () => {
+        // When selected values change, clear the error message
+        // This is so errors aren't shown to the user while they update the
+        // selected values. Note that this should only be called when there are
+        // multiple values that can be selected. onSelectionValidation should be
+        // used whenever the user is done selecting a value or values.
+        setErrorMessage(null);
```

Comment:
**thought:** I wonder if we want to keep this behavior here. I was testing this [here](https://5e1bf4b385e3fb0020b7073c-njsnevcwlb.chromatic.com/?path=/story/packages-dropdown-multiselect--error-from-validation), and I'm not sure if it would make sense to reset the error state to later show it again when the listbox is closed. How do you feel about adding a check here to conditionally reset the error?

 

https://github.com/user-attachments/assets/593e152d-786d-42a0-aa4f-ff8f85bf07ad

Reply:
would calling `validate()` here make sense? Although, that would mean that we are calling `instantValidation` here.

Reply:
After chatting offline, we decided to keep things as they are. Thanks for talking through this @beaesguerra .